### ISSUE TITLE
cpct_memcpy_f8 and cpct_memcpy_f64

### ISSUE THREAD

**Issue by Arnaud6128** [lronaldo/cpctelera#70]
Is possible to speed up the cpct_memcpy with same kind of improvements as cpct_memset_f8 and cpct_memset_f64 functions ?

**Comment by lronaldo** [comment]
It might be possible, but not with same benefit. cpct_memset versions work because they are not copying, but setting a value into memory. That lets the functions store the value in a 16-bits register (like de) and then do several push operations to write the value into memory 16 bits at a time. That costs only 4 microseconds per each 16-bits plus the overhead of the loop. 

However, copying information from one place to another would require poping and pushing, which won't be so fast. However, several games use this technique to copy the whole video memory or part of it faster than with ldir. The technique pays off if you are copying such a great amount of memory, but at a cost: you lose the ability to listen to interrupts, as you need to use SP. 

Summing up, I only see this technique interesting for specific use cases like that, and optimized for that the specific case you wanted. A general memcpy based on push-and-pop won't be so useful, in my opinion. You may gain a few microseconds on big data copies, at a great cost (the space the code would take, plus the use of SP).
